src/modules/user/domain/user.service.ts

```diff
@@ -0,0 +1,38 @@
+import { ClassProvider, Inject, Injectable } from "@nestjs/common";
+import {
+	ServiceCreateUserInput,
+	ServiceUser,
+	UserServiceDefinition,
+	UserServiceName,
+} from "src/modules/user/domain/user.service.types";
+import {
+	RepoUser,
+	UserRepositoryDefinition,
+	UserRepositoryName,
+} from "src/modules/user/repository/user.repository.types";
+
+@Injectable()
+export class UserService implements UserServiceDefinition {
+	constructor(@Inject(UserRepositoryName) private readonly userRepository: UserRepositoryDefinition) {}
+
+	public static Provider(): ClassProvider<UserService> {
+		return {
+			provide: UserServiceName,
+			useClass: UserService,
+		};
+	}
+
+	async createUser({ username, password }: ServiceCreateUserInput): Promise<ServiceUser> {
+		const repoUser: RepoUser = await this.userRepository.createUser({ username: username, password: password });
```

Comment:
So on this line, I am calling the repository, telling it to create a user. I am receiving back a RepoUser (As defined by the repo).

From here, it is our job to properly map it into the Service user (Whatever shape the Service says it's returning to the API)